crml/cennzx/src/lib.rs

```diff
@@ -73,9 +76,21 @@ pub trait Trait: frame_system::Trait {
 	type AssetId: Parameter + Member + AtLeast32BitUnsigned + Default + Copy + Into<u64>;
 	/// The type for asset amounts
 	/// Type for denoting asset balances
-	type Balance: Parameter + Member + AtLeast32BitUnsigned + Default + Copy;
-	/// Something which can provide asset management
-	type AssetSystem: SimpleAssetSystem<AccountId = Self::AccountId, AssetId = Self::AssetId, Balance = Self::Balance>;
+	type Balance: Parameter + Member + AtLeast32BitUnsigned + MaybeSerializeDeserialize + Default + Copy;
+	/// The opaque token type for an imbalance of a particular currency. This is returned by unbalanced operations
+	/// and must be dealt with. It may be dropped but cannot be cloned.
+	type NegativeImbalance: Imbalance<Self::Balance, Opposite = Self::PositiveImbalance>;
+	/// The opaque token type for an imbalance of a particular currency. This is returned by unbalanced operations
+	/// and must be dealt with. It may be dropped but cannot be cloned.
+	type PositiveImbalance: Imbalance<Self::Balance, Opposite = Self::NegativeImbalance>;
+	/// Something which can provide multi currency asset management
+	type MultiCurrency: MultiCurrencyAccounting<
+		AccountId = Self::AccountId,
+		Balance = Self::Balance,
+		CurrencyId = Self::AssetId,
+		NegativeImbalance = Self::NegativeImbalance,
+		PositiveImbalance = Self::PositiveImbalance,
```

Comment:
is it possible to leave these imbalance types unspecified and not on the module trait?

Reply:
I tried it first without the imbalance types but got an error that Imbalance was not implemented as required by the MultiCurrencyAccounting::NegativeImbalance bound. The error was suggesting that I add "where" clause to the cennzx Trait which I can try as an alternative solution.

Reply:
The "where" clause approach seems to create more problems.